SW-1077 Increase Laserhead max. working Temperature

ahmed-mrbeam opened the pull request:
-- Created new folder profiles-->laserheads which contains laserheads yaml properties
-- Extended iobeam-->laserhead_handler to procide a new property laserhead max temprature
-- Added necessary changes to temperature manager
-- in mrbeam plugin init file, reversed the sequence of the init between temprature manager and laserhead handler for dependency reasons
-- Marked the max laserhead temperature found in default printer profile as deprecated

ahmed-mrbeam committed:
SW-1077 Increase Laserhead max. working Temperature
-- Created new folder profiles-->laserheads which contains laserheads yaml properties
-- Extended iobeam-->laserhead_handler to procide a new property laserhead max temprature
-- Added necessary changes to temperature manager
-- in mrbeam plugin init file, reversed the sequence of the init between temprature manager and laserhead handler for dependency reasons
-- Marked the max laserhead temperature found in default printer profile as deprecated

JFSolution commented on octoprint_mrbeam/iobeam/laserhead_handler.py:
```diff
@@ -407,3 +409,65 @@ def _write_laser_heads_file(self, laser_heads_file=None):
+        """
+        Return the current laser head max temperature
+        Returns:
+        Laser head max temp (float)
```
There is a space needed between the description and the returns otherwise it will show wrong in the IDE and the returns have to be tabed once

ahmed-mrbeam replied:
Done, Thanks

JFSolution commented on octoprint_mrbeam/iobeam/laserhead_handler.py:
```diff
@@ -407,3 +409,65 @@ def _write_laser_heads_file(self, laser_heads_file=None):
+        """
+        Loads the current detected laser head related properties and return them
+        Returns:
+            Current laser head properties (dict), (None) otherwise
```
same here, we need a space between description and returns/args

ahmed-mrbeam replied:
Done, Thanks

JFSolution commented on octoprint_mrbeam/iobeam/laserhead_handler.py:
```diff
@@ -407,3 +409,65 @@ def _write_laser_heads_file(self, laser_heads_file=None):
+        laserhead_id = self.get_current_used_lh_model_id()
+
+        # 2. Load the corresponding yaml file and return it's content
+        lh_properties_file_path = os.path.join("/home/pi/oprint/lib/python2.7/site-packages/octoprint_mrbeam",
```
I would not hardcode the path, you could use the path of the package itselfe `plugin._basefolder`

ahmed-mrbeam replied:
Thank you, was looking for this one :). Done

ahmed-mrbeam committed:
SW-895 add-octoprint-version-to-analytics
Addressed review comments:
https://github.com/mrbeam/MrBeamPlugin/pull/1439#discussion_r830868510
https://github.com/mrbeam/MrBeamPlugin/pull/1439#discussion_r830868894
https://github.com/mrbeam/MrBeamPlugin/pull/1439#discussion_r830962573

khaledsherkawi commented on octoprint_mrbeam/iobeam/laserhead_handler.py:
```diff
@@ -407,3 +409,67 @@ def _write_laser_heads_file(self, laser_heads_file=None):
+        Return the current laser head max temperature
+
+        Returns:
+            Laser head max temp (float)
```
Just for a matter of consistency with other comments, I'd rephrase this like other comments in this file:
`float: Laser head max temp`

ahmed-mrbeam replied:
Thanks, Done

khaledsherkawi commented on octoprint_mrbeam/iobeam/laserhead_handler.py:
```diff
@@ -407,3 +409,67 @@ def _write_laser_heads_file(self, laser_heads_file=None):
+        Loads the current detected laser head related properties and return them
+
+        Returns:
+            Current laser head properties (dict), (None) otherwise
```
`dict: current laser head properties, None: otherwise`

ahmed-mrbeam replied:
Thanks, Addressed

khaledsherkawi commented on octoprint_mrbeam/iobeam/laserhead_handler.py:
```diff
@@ -407,3 +409,67 @@ def _write_laser_heads_file(self, laser_heads_file=None):
+            Current laser head properties (dict), (None) otherwise
+
+        """
+        # 1. get the ID of the current laser head
```
I like this ordered list style of comments!

ahmed-mrbeam committed:
Addressed review comments:
https://github.com/mrbeam/MrBeamPlugin/pull/1439#discussion_r831379626
https://github.com/mrbeam/MrBeamPlugin/pull/1439#discussion_r831386295
